### PR TITLE
[agw][lte] Add AGW portion of sms_orc8r service

### PR DESCRIPTION
**Pull request by shaddi**
## Summary
Adds support for SMS-in-MME to the AGW. This adds:

- A new `non_eps_service_control` type, `SMS_ORC8R`, which sets the mme in "SMS-in-MME" mode.
- Adds a new OAI task to handle SMS via orc8r (distinct from existing though similar SGs tasks)
- Defines a new `sms_orc8r` service proto and an implementation of an `sms_orc8r` gateway service in the MME task.

## Test Plan

Verified working with physical setup, `make integ_test`

### CODE REVIEW

**Comment by shaddi** [comment]
Friendly ping @ulaskozat @ssanadhya

File: lte/gateway/c/oai/tasks/grpc_service/SMSOrc8rGatewayServiceImpl.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+SMSOrc8rGatewayServiceImpl::SMSOrc8rGatewayServiceImpl() {}
+
+grpc::Status SMSOrc8rGatewayServiceImpl::SMODownlink(
+    ServerContext* context, const SMODownlinkUnitdata* request, Void* response) {
+  itti_sgsap_downlink_unitdata_t itti_msg;
+  convert_proto_msg_to_itti_sgsap_downlink_unitdata(request, &itti_msg);
+  std::cout << "[DEBUG] "
+            << "Received SMS_ORC8R_DOWNLINK_UNITDATA message from orc8r with IMSI: "
+            << itti_msg.imsi << std::endl;
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/grpc_service/SMSOrc8rGatewayServiceImpl.cpp]: Place these in proper log utility.

File: lte/gateway/c/oai/tasks/grpc_service/SMSOrc8rGatewayServiceImpl.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  return grpc::Status::OK;
+}
+
+grpc::Status SMSOrc8rGatewayServiceImpl::SMOPagingReq(
+    ServerContext* context, const SMOPagingRequest* request, Void* response) {
+  itti_sgsap_paging_request_t itti_msg;
+  convert_proto_msg_to_itti_sgsap_paging_request(request, &itti_msg);
+  std::cout << "[DEBUG] Received SMS_ORC8R_PAGING_REQUEST message"
+               "from orc8r with IMSI: "
+            << itti_msg.imsi << std::endl;
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/grpc_service/SMSOrc8rGatewayServiceImpl.cpp]: place in a proper log utility.

File: lte/gateway/python/scripts/sms_cli.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        description='CLI for SMS',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    # Add subcommands
+    subparsers = parser.add_subparsers(title='subcommands', dest='cmd')
+
+    # Downlink Unitdata
+    downlink_unitdata_parser = subparsers.add_parser(
+            'DU', help="Send downlink unitdata to SMSOrc8rGW service")
+    downlink_unitdata_parser.add_argument('imsi', help='e.g. 123456')
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/sms_cli.py]: in the example it is better to provide a fully qualified IMSI example. If IMSI prefix is needed, also include in the example.

File: lte/gateway/c/oai/tasks/sms_orc8r/sms_orc8r_service_handler.c [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+int handle_sms_orc8r_paging_request(
+    const itti_sgsap_paging_request_t* const sgs_paging_req_pP) {
+  MessageDef* message_p = NULL;
+  int rc                = RETURNok;
+  OAILOG_FUNC_IN(LOG_SMS_ORC8R);
+
+  /* Received Paging Request from orc8r, send it to MME App for further
+   * processing.
+   */
+  message_p = itti_alloc_new_message(TASK_SMS_ORC8R, SGSAP_PAGING_REQUEST);
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/sms_orc8r/sms_orc8r_service_handler.c]: Maybe we should consider renaming this message from SGSAP_PAGING_REQUEST to something more generic.

File: lte/gateway/c/oai/tasks/sms_orc8r/sms_orc8r_service_handler.c [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+#include "sgs_messages_types.h"
+
+int handle_sms_orc8r_downlink_unitdata(
+    const itti_sgsap_downlink_unitdata_t* sgs_dl_unitdata_p) {
+  int rc = RETURNok;
+
+  MessageDef* message_p                              = NULL;
+  itti_sgsap_downlink_unitdata_t* sgs_dl_unit_data_p = NULL;
+
+  message_p = itti_alloc_new_message(TASK_SMS_ORC8R, SGSAP_DOWNLINK_UNITDATA);
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/sms_orc8r/sms_orc8r_service_handler.c:36]: consider changing this message type and definitions to a more generic type. doesn't have to be this PR.

File: lte/protos/sms_orc8r.proto [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+package magma.lte;
+option go_package = "magma/lte/cloud/go/protos";
+
+service SMSOrc8rService {
+    rpc SMOUplink (SMOUplinkUnitdata) returns (magma.orc8r.Void) {}
+}
+
+service SMSOrc8rGatewayService {
+    rpc SMODownlink (SMODownlinkUnitdata) returns (magma.orc8r.Void) {}
+    rpc SMOPagingReq (SMOPagingRequest) returns (magma.orc8r.Void) {}
```
Comment (ulaskozat) [inline lte/protos/sms_orc8r.proto]: why do we need a separate RPC for paging? Should not SMODownlink rpc call trigger paging and message delivery?

**Review by ulaskozat — CHANGES_REQUESTED** [review]
Let's have the DL SMS data trigger paging if UE is in idle mode rather than defining an rpc call for paging.

File: lte/gateway/python/scripts/sms_cli.py (hunk 3 above)
Comment (shaddi) [reply lte/gateway/python/scripts/sms_cli.py]: I'll put a full example in, but as implemented this does not require an IMSI prefix.

**Review by ulaskozat — APPROVED** [review]
LGTM. Please make sure that paging is handled as we expected before merging.

**Comment by shaddi** [comment]
I've verified paging works as intended during idle mode delivery, and I've opened #2945 to track the suggestion on the sgsap types.

File: lte/gateway/c/oai/tasks/grpc_service/SMSOrc8rGatewayServiceImpl.cpp [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+using grpc::ServerContext;
+
+namespace magma {
+
+SMSOrc8rGatewayServiceImpl::SMSOrc8rGatewayServiceImpl() {}
+
+grpc::Status SMSOrc8rGatewayServiceImpl::SMODownlink(
+    ServerContext* context, const SMODownlinkUnitdata* request, Void* response) {
```
Comment (ardzoht) [inline lte/gateway/c/oai/tasks/grpc_service/SMSOrc8rGatewayServiceImpl.cpp:48]: I think these can be passed by const reference?

Comment (ardzoht) [reply lte/gateway/c/oai/tasks/grpc_service/SMSOrc8rGatewayServiceImpl.cpp:48]: actually discard my comment, these are used by the gRPC servicer and it expects the params to be passed as pointer

**Review by ardzoht — APPROVED** [review]
lgtm

**Pull request merged**